scripts/tests/twister/test_harness.py

```diff
@@ -34,6 +43,348 @@ def process_logs(harness, logs):
+    console.GCOV_END = "COVERAGE_END"
+    console.record = {"regex": "RESULT: (.*)"}
+    console.fieldnames = []
+    console.recording = []
+    console.regex = ["regex1", "regex2"]
+    console.id = "test_case_1"
+    instance.get_case_or_create('test_case_1')
+    instance.testsuite.id = "test_suite_1"
+    console.next_pattern = 0
+    console.ordered = ordered_val
+    line = line
+    console.handle(line)
+
+    line1 = "pattern1"
+    line2 = "pattern2"
+    console.handle(line1)
+    console.handle(line2)
+    assert console.state == exp_state
+    with pytest.raises(Exception):
+        console.handle(line)
+        assert logger.error.called
+    assert console.fault == exp_fault
+    assert console.capture_coverage == exp_capture
+
+TEST_DATA_2 = [("serial_pty", 0), (None, 0),(None, 1)]
+@pytest.mark.parametrize(
+   "pty_value, hardware_value",
+   TEST_DATA_2,
+   ids=["hardware pty", "hardware", "non hardware"]
+)
+
+def test_pytest__generate_parameters_for_hardware(pty_value, hardware_value):
+    #Arrange
+    mock_platform = mock.Mock()
+    mock_platform.name = "mock_platform"
+    mock_testsuite = mock.Mock(id = 'id', testcases = [])
+    mock_testsuite.name = "mock_testsuite"
+    instance = TestInstance(testsuite=mock_testsuite, platform=mock_platform, outdir="")
+    handler = mock.Mock()
+    handler.instance = instance
+    hardware = mock.Mock()
+    hardware.serial_pty = pty_value
+    hardware.serial = 'serial'
+    hardware.baud = 115200
+    hardware.runner = "runner"
+    options = handler.options
+    options.west_flash = "args"
+    hardware.probe_id = '123'
+    hardware.product = 'product'
+    hardware.pre_script = 'pre_script'
+    hardware.post_flash_script = 'post_flash_script'
+    hardware.post_script = 'post_script'
+    pytest_test = Pytest()
+
+    #Act
+    if hardware_value == 0:
+        handler.get_hardware.return_value = hardware
+        command = pytest_test._generate_parameters_for_hardware(handler)
+    else:
+        handler.get_hardware.return_value = None
+
+    #Assert
+    if hardware_value == 1:
+        with pytest.raises(PytestHarnessException) as exinfo:
+            pytest_test._generate_parameters_for_hardware(handler)
+        assert str(exinfo.value) == 'Hardware is not available'
+    else:
+        assert '--device-type=hardware' in command
+        if pty_value == "serial_pty":
+            assert '--device-serial-pty=serial_pty' in command
+        else:
+            assert '--device-serial=serial' in command
+            assert '--device-serial-baud=115200' in command
+        assert '--runner=runner' in command
+        assert '--west-flash-extra-args=args' in command
+        assert '--device-id=123' in command
+        assert '--device-product=product' in command
+        assert '--pre-script=pre_script' in command
+        assert '--post-flash-script=post_flash_script' in command
+        assert '--post-script=post_script' in command
```

Comment:
These could be gathered in some way, like:
```python
    else:
        cmd_list = ['--device-type=hardware',
                    '--runner=runner',
                    '--west-flash-extra-args=args',
                    '--device-id=123',
                    '--device-product=product',
                    '--pre-script=pre_script',
                    '--post-flash-script=post_flash_script',
                    '--post-script=post_script']
                           
        if pty_value == "serial_pty":
            cmd_list += ['--device-serial-pty=serial_pty']
        else:
            cmd_list += ['--device-serial=serial', '--device-serial-baud=115200']
                           
        assert all(cmd in command for cmd in cmd_list)
```

However, the result is worse when it comes to analysing which command is actually missing, so I am only posting this here as a curiosity.